monai/data/box_utils.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from typing import Sequence, Union
+
+import numpy as np
+import torch
+
+SUPPORT_MODE = ["xxyy", "xxyyzz", "xyxy", "xyzxyz", "xywh", "xyzwhd"]
+STANDARD_MODE = ["xxyy", "xxyyzz"]  # [2d_mode, 3d_mode]
+
+# TO_REMOVE = 0 if in 'xxyy','xxyyzz' mode, the bottom-right corner is not included in the box,
+#      i.e., when x_min=1, x_max=2, we have w = 1
+# TO_REMOVE = 1  if in 'xxyy','xxyyzz' mode, the bottom-right corner is included in the box,
+#       i.e., when x_min=1, x_max=2, we have w = 2
+TO_REMOVE = 0  # x_max-x_min = w -TO_REMOVE
+
+"""
+The following variables share the same definition across the functions in this file.
+Args:
+    bbox: Nx4 or Nx6 torch tensor
+    mode: choose from SUPPORT_MODE. If mode is not given, these funcs will assume mode is STANDARD_MODE
+    image_size: Length of 2 or 3. Data format is list, or np.ndarray, or tensor of int
+"""
+
+
+def check_support_mode(mode):
+    """
+    Check if the mode is supported
+    """
+    if mode not in SUPPORT_MODE:
```

Comment:
there's a `look_up_option ` utility, please make use of it... https://github.com/Project-MONAI/MONAI/blob/3c6ee694d5e7fd6d9da2c65ed1b83030488da064/monai/utils/module.py#L47-L74

there are also some other utilities, might be useful https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/misc.py#L29-L48